client/src/components/HistoryViewer/HistoryViewerVersionDetail.js

```diff
@@ -164,19 +165,13 @@ HistoryViewerVersionDetail.propTypes = {
   schemaUrl: PropTypes.string.isRequired,
   ToolbarComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
   version: versionType,
-  compare: PropTypes.oneOfType([
-    PropTypes.shape({
-      versionFrom: versionType,
-      versionTo: versionType,
-    }),
-    PropTypes.bool,
-  ]),
```

Comment:
I considered doing this in my PR but in this particular component, this compare prop is different. All the others just have `versionFrom` and `versionTo` as numbers, but this has them as `versionType`s. It bugs me, but I can't really think of a decent solution right now.

Reply:
Oh true! We should refactor it so everything uses IDs then

Reply:
It was set this way as this is how it existing logic for displaying the version worked (c.f. `version.Version` references).
Not arguing against this, probably more pointing out that if we're going to refactor to make comparison version passing to be all ID references, then we should probably do the same for normal `currentVersion` references, as they were the model for the current status (work the same way).

Reply:
We could go the other way and make everything store objects

Reply:
We could go the other way and make everything store objects

Reply:
I think this should be split into a separate issue, as it would require relocating a fair bit of logic with care in keeping separation of concerns and coupling to a minimum. That in itself I feel warrants it's own piece of work.

Reply:
Fair call. It was intended as a low impact refactor initially but I didn't read the context properly. I'll remove it now.